src/ripple/ledger/ApplyView.h

```diff
@@ -66,6 +63,32 @@ operator&(ApplyFlags const& lhs,
             static_cast<int>(rhs));
 }
 
+inline
+ApplyFlags
+operator~(ApplyFlags const& flags)
+{
+    return static_cast<ApplyFlags>(
+        ~static_cast<int>(flags));
```

Comment:
I agree. It probably makes sense for ripple to have an `underlying` free function which gets the underlying type of an enum class to avoid the mouthful wording at each call site.

Reply:
Not so sure about making an `underlying()` free template function.  By my reckoning it would save approximately 12 characters per invocation.  Speaking only for myself, if I see a standard library method when reading the source code then I know exactly what I'm getting and where to find the documentation.  If I see anything else then I have to chase it down to understand it.  The character count savings may not be worth it.  But that's just my perspective...

Reply:
I'm not sold that this is a good idea, but 
```
template<class T>
constexpr
auto to_underlying(T value)
{
    return static_cast<std::underlying_type_t<T>>(value);
}
```
reduces `static_cast<std::underlying_type_t<ApplyFlags>>(flags)` to `to_underlying(flags)`, as savings of 34 characters per call.

Reply:
You have a point.  I was thinking of an equivalent of `std::underlying_type_t<>` that would return a type.  I think that would give us (only) a 12 character savings.

The 34 character savings per call is interesting.  However, since that transformation is often not a good idea, I'm not sure we should make it that easy.  I'm open to discussion.